text/0033-private-testnet-guide.md

```diff
@@ -0,0 +1,425 @@
+
+```sh
+curl -X POST --data "wallet-id=wallet" --data "seed=$SEED" http://localhost:8000/start
+sleep 5
+curl -H "X-Wallet-Id: wallet" http://localhost:8000/wallet/address
+```
+
+Write down this address, we will use it in the next section.
+
+Finally, stop the hathor-wallet-headless.
+
+```sh
+docker stop wallet-headless
+docker rm wallet-headless
+```
+
+### Important Note
+
+Note that we used a server from our public testnet in the config above. This was done because the hathor-wallet-headless needs it to be able to start a wallet and generate an address for it in the following commands. From now on, we won't need to do it anymore.
+
+However, to make things easier, we use the same address prefix as the public testnet in the next sections when configuring our private testnet, since we used it to generate an address. Otherwise, we would need to make an address translation from one prefix to another.
+
+## Creating the genesis block and transactions
+Now that we have an address, we will use it to create our genesis block and our two initial transactions.
+
+Run this code in a python shell inside our hathor-core Docker image, to create the output script for the genesis. Keep note of the returned value to be used in the next steps as the GENESIS_OUTPUT_SCRIPT.
+
+You need to replace `<address>` with the address you got in previous steps.
+
+```sh
+export GENESIS_OUTPUT_SCRIPT=$(docker run -it --entrypoint python  hathornetwork/hathor-core -c "
+from hathor.transaction.scripts import P2PKH
+import base58
+
+address = base58.b58decode('<address>')
+
+output = P2PKH.create_output_script(address=address).hex()
+print(output, end='')
+")
+```
+
+Now to configure the parameter of our new network, we create a new file in `~/hathor-private-tutorial/conf/privnet.py`, with the following content. Note that we are using the `GENESIS_OUTPUT_SCRIPT` variable from the previous command.
+
+```sh
+mkdir -p ~/hathor-private-tutorial/conf
+
+cat << EOF > ~/hathor-private-tutorial/conf/privnet.py
+from hathor.conf.settings import HathorSettings
+
+SETTINGS = HathorSettings(
+    P2PKH_VERSION_BYTE=b'\x49',
+    MULTISIG_VERSION_BYTE=b'\x87',
+    NETWORK_NAME='privnet-testnet',
+    BOOTSTRAP_DNS=[],
+    # Genesis stuff
+    GENESIS_OUTPUT_SCRIPT=bytes.fromhex("$GENESIS_OUTPUT_SCRIPT"),
+    GENESIS_TIMESTAMP=1632426140,
+    MIN_TX_WEIGHT_K=0,
+    MIN_TX_WEIGHT_COEFFICIENT=0,
+    MIN_TX_WEIGHT=8,
+    REWARD_SPEND_MIN_BLOCKS=20
+)
+EOF
+```
+
+Run the python shell again, with additional parameters to use our newly created settings file, and passing a script to mine the genesis block and 2 initial transactions. Keep note of the printed outputs, they will be used next.
+
+```sh
+docker run -it --entrypoint python  -v ~/hathor-private-tutorial/conf:/privnet/conf --env PYTHON_PATH=/privnet/conf --env HATHOR_CONFIG_FILE=privnet.conf.privnet hathornetwork/hathor-core -c "
```

Comment:
Because of `PYTHON_PATH=/privnet/conf` it should be possible to use just `HATHOR_CONFIG_FILE=privnet`, otherwise if `HATHOR_CONFIG_FILE=privnet.conf.privnet` is working, then setting `PYTHON_PATH` would be redundant. I'd still prefer using `HATHOR_CONFIG_FILE=privnet`, it might set the example of how to put it in any folder since the other approach has a lot of other limitations (`/privnet-conf` with `HATHOR_CONFIG_FILE=privnet-conf.privnet` would not work, for example).

Reply:
This doesn't work for me with `PYTHON_PATH=/privnet/conf` and `HATHOR_CONFIG_FILE=privnet`:

```
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/usr/local/lib/python3.7/site-packages/hathor/transaction/__init__.py", line 16, in <module>
    from hathor.transaction.base_transaction import (
  File "/usr/local/lib/python3.7/site-packages/hathor/transaction/base_transaction.py", line 46, in <module>
    from hathor.transaction.transaction_metadata import TransactionMetadata
  File "/usr/local/lib/python3.7/site-packages/hathor/transaction/transaction_metadata.py", line 20, in <module>
    from hathor.util import practically_equal
  File "/usr/local/lib/python3.7/site-packages/hathor/util.py", line 30, in <module>
    settings = HathorSettings()
  File "/usr/local/lib/python3.7/site-packages/hathor/conf/get_settings.py", line 40, in HathorSettings
    return module.SETTINGS  # type: ignore
AttributeError: module 'privnet' has no attribute 'SETTINGS'
```

It considers the module `privnet` to be `/privnet`

Reply:
But removing `PYTHON_PATH` and having just `HATHOR_CONFIG_FILE=privnet.conf.privnet` does work.